pages/History.qml

```diff
@@ -34,342 +34,1519 @@ import moneroComponents.WalletManager 1.0
+            root.model.amountFromFilter = -1;
+            root.model.amountToFilter = -1;
+            root.model.directionFilter = TransactionInfo.Direction_Both;
+        }
+    }
+
+    function updateFilter(){
+        // applying filters
+        root.txData = JSON.parse(JSON.stringify(root.txModelData)); // deepcopy
+
+        var fromDate = fromDatePicker.currentDate.getTime() / 1000;
+        var toDate = toDatePicker.currentDate.getTime() / 1000;
+
+        var txs = [];
+        for (var i = 0; i < root.txData.length; i++){
+            var item = root.txData[i];
+            var matched = "";
+
+            // daterange filtering
+            if(item.timestamp < fromDate || item.timestamp > toDate){
+                continue;
+            }
+
+            // search string filtering
+            if(root.sortSearchString == null || root.sortSearchString === ""){
+                txs.push(root.txData[i]);
+                continue;
+            }
+
+            if(root.sortSearchString.length >= 1){
+                if(item.amount && item.amount.toString().startsWith(root.sortSearchString)){
+                    txs.push(item);
+                } else if(item.address !== "" && item.address.startsWith(root.sortSearchString)){
+                    txs.push(item);
+                } else if(item.blockheight.toString().startsWith(root.sortSearchString)) {
+                    txs.push(item);
+                } else if (item.hash.startsWith(root.sortSearchString)){
+                    txs.push(item);
                 }
             }
         }
+
+        root.txData = txs;
+        root.txCount = root.txData.length;
+
+        root.updateSort();
+        root.updateDisplay(root.txOffset, root.txMax);
+    }
+
+    function updateSort(){
+        // applying sorts
+        root.txOffset = 0;
+        root.txData.sort(function(a, b) {
+            return a[root.sortBy] - b[root.sortBy];
+        });
+
+        if(root.sortDirection)
+            root.txData.reverse();
+
+        root.updateDisplay(root.txOffset, root.txMax);
+    }
+
+    function updateDisplay(tx_offset, tx_max, auto_collapse) {
+        if(typeof auto_collapse === 'undefined') auto_collapse = false;
+        txListViewModel.clear();
+
+        // limit results as per tx_max (root.txMax)
+        var txs = root.txData.slice(tx_offset, tx_offset + tx_max);
+
+        // make first result on the first page collapsed by default
+        if(auto_collapse && root.txPage === 1 && txs.length > 0 && (root.sortSearchString == null || root.sortSearchString === ""))
+            root.txDataCollapsed.push(txs[0]['hash']);
+
+        // populate listview
+        for (var i = 0; i < txs.length; i++){
+            txListViewModel.append(txs[i]);
+        }
+
+        root.updateHistoryStatusMessage();
+
+        // determine pagination button states
+        var count = txData.length;
+        if(count <= root.txMax) {
+            paginationPrev.enabled = false
```

Comment:
```suggestion
        informationPopup.title = qsTr("Transaction details");
```

Reply:
Weird glitch here on github. Should be line 1496 needs a qsTr.

Reply:
:+1: